### PR TITLE
fix(curriculum): cat photo app workshop update step 2

### LINKED ISSUE #56307: Update content and tests for step 2 of cat photo app workshop in frontend cert
### Describe the Issue

We recently updated the guidelines for writing assertions here 
https://contribute.freecodecamp.org/how-to-work-on-coding-challenges/#writing-tests

For consistency and matching what is in the docs, we should slowly start to update the tests for the workshops and labs in the new frontend cert.

For this issue, just focus on updating the tests for step 2 of the cat photo app workshop.
We don't want to have a large PR that updates a whole bunch of tests in a bunch of steps and makes it harder to review. 

https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6690b9626a66f51257ae5a91.md

Those tests should be updated, where appropriate, to use specific methods like `assert.equal`, `assert.exists`, `assert.match`, etc instead of just `assert`.

Here is the complete API reference 
https://www.chaijs.com/api/assert/#method_exists

Just focus on the new frontend cert and don't worry about updating the RWD cat photo app since that will become legacy anyway.

Also, update the hint to the following here

```md
Your `h1` element should have a closing tag. Closing tags have this syntax: `</elementName>`.
```

And update this hint text to use backticks consistently:

```md
Your `h2` element's text should be `Cat Photos`. Only place the text `Cat Photos` between the opening and closing `h2` tags.
```

### Affected Page

https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6690b9626a66f51257ae5a91.md

### Your code

see explanation above

### Expected behavior

see explanation above

### Screenshots

_No response_

### System

- Device: [e.g. iPhone 6, Laptop]
- OS: [e.g. iOS 14, Windows 10, Ubuntu 20.04]
- Browser: [e.g. Chrome, Safari]
- Version: [e.g. 22]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by gagan-bhullar-tech**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56307

### CODE REVIEW

**Comment by ilenia-magoni** [comment]
the title of the project is cat photo app, please change to that so that it is understandable for what project this issue is for

**Comment by gagan-bhullar-tech** [comment]
@ilenia-magoni updated title

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6690b9626a66f51257ae5a91.md [hunk 1]
```diff
@@ -29,45 +29,43 @@ Below the `h1` element, add an `h2` element with this text:
...
+assert.match(code, /<\/h1\>/);
 ```
 
 You should only have one `h1` element. Remove the extra.
 
 ```js
-assert(
-  document.querySelector('h1') && document.querySelectorAll('h1').length === 1
-);
+assert.equal(document.querySelectorAll('h1')?.length, 1);
```
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6690b9626a66f51257ae5a91.md:44]:
```suggestion
assert.lengthOf(document.querySelectorAll('h1'), 1);
```

File: curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6690b9626a66f51257ae5a91.md [hunk 2]
```diff
@@ -29,45 +29,43 @@ Below the `h1` element, add an `h2` element with this text:
...
 ```js
-assert(document.querySelector('h2'));
+assert.exists(document.querySelector('h2'));
 ```
 
 Your `h2` element should have a closing tag. Closing tags have a `/` just after the `<` character.
 
 ```js
-assert(code.match(/<\/h2\>/));
+assert.exists(code.match(/<\/h2\>/));
```
Comment (gikf) [inline curriculum/challenges/english/25-front-end-development/workshop-cat-photo-app/6690b9626a66f51257ae5a91.md:62]:
```suggestion
assert.match(code, /<\/h2\>/);
```

**Review by gikf — APPROVED** [review]

**Review by jdwilkin4 — APPROVED** [review]

**Pull request merged**